pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +194,74 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
 	}
 
+	results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions = instanceTypesOptionsFilteredByPrice
+	return Command{
+		candidates:   candidates,
+		replacements: results.NewNodeClaims,
+	}, nil
+}
+
+// Compute command to execute spot-to-spot consolidation if:
+//  1. The SpotToSpotConsolidation feature flag is set to true.
+//  2. For single-node consolidation:
+//     a. There are at least 15 cheapest instance type replacement options to consolidate.
+//     b. The current candidate is NOT part of the first 15 cheapest instance types inorder to avoid repeated consolidation.
+func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate, results *pscheduling.Results,
+	candidatePrice float64) (Command, error) {
+
+	// Spot consolidation is turned off.
+	if !options.FromContext(ctx).FeatureGates.SpotToSpotConsolidation {
+		if len(candidates) == 1 {
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node")...)
+		}
+		logging.FromContext(ctx).Debugf("SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node.")
+		return Command{}, nil
+	}
+
+	// All possible replacements for the current candidate filtered by spot offerings
+	instanceTypeOptionsWithSpotOfferings :=
```

Comment:
There's a lot of variables that we are storing things in in this codeblock and reading through it the first time I was struggling to reason about when we were using certain variables and when we were using others. I wonder if there's a way to collapse a lot of this so that we don't have to reason about so many different variables.

Reply:
Reduced the number of variables from 3 to 1 and refactored it to segregate the single-node and multi-node logic.